test/quantization/fx/test_model_report_fx.py

```diff
@@ -431,3 +435,276 @@ def forward(self, x):
+        def __init__(self):
+            super().__init__()
+            self.obs1 = ModelReportObserver()
+            self.mod1 = SingleLayerLinearModel()
+            self.obs2 = ModelReportObserver()
+            self.fc1 = torch.nn.Linear(5, 5).to(dtype=torch.float)
+            self.relu = torch.nn.ReLU()
+
+        def forward(self, x):
+            x = self.obs1(x)
+            x = self.mod1(x)
+            x = self.obs2(x)
+            x = self.fc1(x)
+            x = self.relu(x)
+            return x
+
+    def run_model_and_common_checks(self, model, ex_input, num_epochs, batch_size):
+        # split up data into batches
+        split_up_data = torch.split(ex_input, batch_size)
+        for epoch in range(num_epochs):
+            # reset all model report obs
+            model.apply(
+                lambda module: module.reset_batch_and_epoch_values()
+                if isinstance(module, ModelReportObserver)
+                else None
+            )
+
+            # quick check that a reset occurred
+            self.assertEqual(
+                getattr(model, "obs1").average_batch_activation_range,
+                torch.tensor(float(0)),
+            )
+            self.assertEqual(
+                getattr(model, "obs1").epoch_activation_min, torch.tensor(float("inf"))
+            )
+            self.assertEqual(
+                getattr(model, "obs1").epoch_activation_max, torch.tensor(float("-inf"))
+            )
+
+            # loop through the batches and run through
+            for index, batch in enumerate(split_up_data):
+
+                num_tracked_so_far = getattr(model, "obs1").num_batches_tracked
+                self.assertEqual(num_tracked_so_far, index)
+
+                # get general info about the batch and the model to use later
+                batch_min, batch_max = torch.aminmax(batch)
+                current_average_range = getattr(
+                    model, "obs1"
+                ).average_batch_activation_range
+                current_epoch_min = getattr(model, "obs1").epoch_activation_min
+                current_epoch_max = getattr(model, "obs1").epoch_activation_max
+
+                # run input through
+                model(ex_input)
+
+                # check that average batch activation range updated correctly
+                correct_updated_value = (
+                    current_average_range * num_tracked_so_far + (batch_max - batch_min)
+                ) / (num_tracked_so_far + 1)
+                self.assertEqual(
+                    getattr(model, "obs1").average_batch_activation_range,
+                    correct_updated_value,
+                )
+
+                if current_epoch_max - current_epoch_min > 0:
+                    self.assertEqual(
+                        getattr(model, "obs1").get_batch_to_epoch_ratio(),
+                        correct_updated_value / (current_epoch_max - current_epoch_min),
+                    )
+
+    """Case includes:
```

Comment:
I'm still wondering if we can reduce the number of tests, I feel to start with we can do
1). edge case tests where we call `calculate_qparams` before we run through any data
2). normal case when we feed in some tensor and the result is correct

we can add more tests when needed in the future

Reply:
Got rid of one of the less useful test cases. All other cases are either edge cases or a run multiple shapes and dimensions case.